cfme/networks/topology.py

```diff
@@ -0,0 +1,232 @@
+                        elements_obj.append(element)
+        return elements_obj
+
+    @property
+    def children(self):
+        elements_obj = []
+        for lines_obj in [lines_obj for lines_obj in self.object.lines_obj()
+                          if lines_obj.x1 == self.x and lines_obj.y1 == self.y]:
+            for element in self.object.elements_obj():
+                if element.x == lines_obj.x2 and element.y == lines_obj.y2:
+                    elements_obj.append(element)
+        return elements_obj
+
+    def double_click(self):
+        self.obj.browser.double_click(self.element)
+
+    def is_displayed(self):
+        try:
+            return self.element.is_displayed()
+        except Exception:
+            return False
+
+
+@attr.s
+class TopologyElementCollection(BaseCollection):
+    """Collection object for elements in topology"""
+    ENTITY = TopologyElement
+
+    def instantiate(self, *args, **kwargs):
+        elem = super(TopologyElementCollection, self).instantiate(*args, **kwargs)
+        assert elem.element, 'Element should not be None'
+        element_data = re.search('Name: (.*) Type: (.*) Status: (.*)', elem.element.text)
+        assert len(element_data.groups()) == 3, 'Topology element doesnt have name, type or status'
+        elem.name = element_data.group(1)
+        elem.type = element_data.group(2)
+        elem.status = element_data.group(3)
+        elem.x = round(float(elem.element.get_attribute('cx')), 1)
+        elem.y = round(float(elem.element.get_attribute('cy')), 1)
+        return elem
+
+    def filter(self, obj):
+        elements = obj.browser.elements(obj.view.ELEMENTS)
+        return [self.instantiate(element, obj) for element in elements]
+
+
+@attr.s
+class TopologyLine(BaseEntity):
+    """ Class representing line connecting 2 nodes in Topology
+        Consists of Browser element
+    """
+    element = attr.ib()
+
+
+@attr.s
+class TopologyLineCollection(BaseCollection):
+    """Collection object for lines in topology"""
+    ENTITY = TopologyLine
+
+    def instantiate(self, *args, **kwargs):
+        line = super(TopologyLineCollection, self).instantiate(*args, **kwargs)
+        assert line.element, 'Element should not be None'
+        line.connection = line.element.get_attribute('class')
+        line.x1 = round(float(line.element.get_attribute('x1')), 1)
+        line.x2 = round(float(line.element.get_attribute('x2')), 1)
+        line.y1 = round(float(line.element.get_attribute('y1')), 1)
+        line.y2 = round(float(line.element.get_attribute('y2')), 1)
+        return line
+
+    def filter(self, topology):
+        lines = topology.browser.elements(topology.view.LINES)
+        return [self.instantiate(element=line) for line in lines]
+
+
+@attr.s
+class Topology(BaseEntity):
+    "Class represents SDN topology"
+    elements_obj = attr.ib(default=[])
+    lines_obj = attr.ib(default=[])
+    legends_obj = attr.ib(default=[])
```

Comment:
**REQUIRED**
It's very not recommended to use mutable types such as `list` as an argument. Use `None` here.